### ISSUE TITLE
FEAT - Double quadratic datafit (quad-quad)

### ISSUE THREAD

**Issue by tomaszkacprzak** [scikit-learn-contrib/skglm#272]
## Description of the feature

In many applications there is a need to have an asymmetric loss that weights negative and positive residuals differently. A good function for this is double quadratic loss (quad-quad):

$L (\epsilon, \alpha) = 2 · ( \alpha + (1 − 2\alpha) · 1[\epsilon > 0] ) · \| \epsilon \|_2^2$

where $\epsilon =  Xw - y$ is the residual, $\alpha$ controls the degree of asymmetry and  $1[·]$ is an indicator function. For $\alpha=0.5$ this function is equivalent to Quadratic datafit.

Would it be possible to add DoubleQuadratic datafit and have it compatible with the Lasso and other penalties?

## Considered alternatives

Another possibility is to use Pinball loss.

**Comment by Badr-MOUFAD** [comment]
Also I believe it should be $1 / 2$ instead of $2$ to get ride of the coefficient that pops up after differentiation

$L (\epsilon, \alpha) = \frac{1}{2} \sum_i ( \alpha + (1 − 2\alpha) · 1[\epsilon_i > 0] ) \epsilon_i^2$

**Comment by tomaszkacprzak** [comment]
Hi @mathurinm thank you for a fast reply. Indeed there was a mistake in my equation, the one you gave should be right. I can try a PR. I could start with a Quadratic and modify all the functions. Do you expect any difficulties for optimisation?

**Comment by mathurinm** [comment]
Yes that sounds like a plan, and no I don't foresee any difficulty!

**Comment by mathurinm** [comment]
@tomaszkacprzak could you also please let us know in which context you use skglm, and what it brings you compared to alternatives?

**Comment by mathurinm** [comment]
Hi @tomaszkacprzak, any news on this?
